Fix #55: Save qps as string

bserdar opened the pull request.

jewzaam commented:
Isn't there still a problem with saving?  Thought `putProjection` would need updated as well to save the projection, sort, and query as string instead of bson.

jewzaam commented:
Would be good to have a unit test added for the case that query (part of association) contains a key with `$` in it and try persisting the metadata.

bserdar commented:
You are correct. Fixed now.

On Thu, Jan 15, 2015 at 1:22 PM, Naveen Malik <notifications@github.com>
wrote:

> Isn't there still a problem with saving? Thought putProjection would need
> updated as well to save the projection, sort, and query as string instead
> of bson.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-mongo/pull/56#issuecomment-70154688>
> .
>

bserdar commented:
Unit test is also added.

On Thu, Jan 15, 2015 at 1:27 PM, Burak Serdar <bserdar@ieee.org> wrote:

> You are correct. Fixed now.
>
> On Thu, Jan 15, 2015 at 1:22 PM, Naveen Malik <notifications@github.com>
> wrote:
>
>> Isn't there still a problem with saving? Thought putProjection would
>> need updated as well to save the projection, sort, and query as string
>> instead of bson.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/lightblue-platform/lightblue-mongo/pull/56#issuecomment-70154688>
>> .
>>
>
>

jewzaam commented on metadata/src/test/java/com/redhat/lightblue/metadata/mongo/MongoMetadataTest.java:
```diff
@@ -173,6 +173,25 @@ public void createMdTest() throws Exception {
+        JSONMetadataParser parser = new JSONMetadataParser(extensions, new DefaultTypes(), new JsonNodeFactory(true));
+
+        // get JsonNode representing metadata
+        JsonNode jsonMetadata = AbstractJsonNodeTest.loadJsonNode(getClass().getSimpleName() + "-qps-andquery.json");
```
Test file missing.  I'll merge this anyway with this test ignored so builds are not broken.  It's blocking work @derek63 is doing for hystrix.

jewzaam replied:
Issue to fix the test added: https://github.com/lightblue-platform/lightblue-mongo/issues/57
